utils/systemd/monerod.service

```diff
@@ -1,37 +1,87 @@
+### Sample unit file for running monerod as a systemd service ###
+
+### For more information about the options specified in the [Unit] and [Install] sections,
+### please see the systemd.unit(5) man page.
+
 [Unit]
-Description=Monero Full Node
+
+Description=Monero Daemon
 After=network.target
 
+### If monerod is configured to use Tor via "--tx-proxy" and, optionally, "--anonymous-inbound",
+### and if Tor is installed and running as a systemd service,
+### then replace the above "After=network.target" directive with the following two directives:
+
+# After=network.target tor.service
+# Wants=tor.service
+
+### The following [Install] section allows monero.service to be enabled via "systemctl enable",
+### which in turn causes monero.service to be automatically started at system startup.
+### The "Alias=monerod.service" directive allows monero.service to be invoked as monerod.service in various systemctl commands
+### if monero.service is enabled via "systemctl enable".
+
+[Install]
+
+Alias=monerod.service
+WantedBy=multi-user.target
+
+### For more information about the options specified in the [Service] section,
+### please see the systemd.service(5) and systemd.exec(5) man pages.
+
 [Service]
+
+Type=simple
+
+### The following "ExecStart=" line assumes that:
+### (a) the monerod binary has been copied to the /usr/local/bin directory and is owned by the "monero" system user and group,
+### (b) the /etc/monero directory exists and is owned by the "monero" system user and group, and
+### (c) the monerod.conf configuration file exists, is located in /etc/monero, and is owned by the "monero" system user and group.
+
+ExecStart=/usr/local/bin/monerod --config-file /etc/monero/monerod.conf --non-interactive
+RestartSec=60
+TimeoutStopSec=300
```

Comment:
This seems wrong/unduly long. The default (which is 90 sec) should be more than adequate.

Reply:
The decision to set `TimeoutStopSec` to five minutes was inspired by Bitcoin's decision to set theirs to 10 minutes. See their rationale in commit [`99cebc9`](https://github.com/bitcoin/bitcoin/commit/99cebc922cba1590ea663d6ef3109c0cf2960b9d#diff-6ca46037cb8f71de1d76344d1892f28d00ccd5b9e09725842809d288293abbbf).

I found 10 minutes to be overkill, so I halved it to five minutes.

The `systemd.service(5)` man page does not specify a recommended value for the `TimeoutStopSec=` directive. Similarly, the `systemd-system.conf(5)` man page does not specify a recommended value for the `DefaultTimeoutStopSec=` directive.

Reply:
Simply copying Bitcoin's choice is not a good enough rational. These are different daemons with different shutdown actions. Normally the monerod daemon shutdown is is quick, no more than a few seconds. I'd expect anyone editing this file actually tests what an appropriate timeout should be rather than finger in the air guesswork.

Reply:
Bitcoind's ordinary shutdown duration is also quick, and yet they saw fit to set `TimeoutStopSec=` to 10 minutes. But I digress.

The timeout will not be reached under ordinary circumstances. But in the event of an extraordinary situation where monerod takes much longer to gracefully stop, it is better to have a longer timeout than a shorter one. If the user does not want to wait the whole five minutes, then they can either manually send a `SIGKILL` or edit the `monerod.service` file to shorten the timeout duration for all future runs. After all, this file is a sample file. The user is free to adapt the file to their own needs.

Reply:
Fair enough. How about `RestartSec`?

Reply:
It gives systemd time to restore the system to a pre-`monerod.service` state before the next run, after it receives a non-zero exit signal from `monerod` (because `Restart=` is set to `on-failure`).

The default `RestartSec=` value of 100 milliseconds is too short, especially on slower systems. A repeat of the memory exhaustion attack would also mean that the system may not have enough time to flush its swap memory before systemd relaunches monerod. One minute is long enough for systemd to do what it needs to do on slower systems, but not too long that it would be a significant annoyance.

Again, this is merely a sample unit file. The user is free to set a shorter `RestartSec=` value if they see fit. One minute between runs in `monerod.service`'s case is a good default for most systems.